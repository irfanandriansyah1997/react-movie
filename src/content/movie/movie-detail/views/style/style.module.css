.detail {
    display: flex;
    margin: 30px 0;
    padding-top: 70px;
}

.detail img {
    width: 200px;
    height: 300px;
    object-fit: cover;
    border-radius: 10px;
}

.detail table {
    width: calc(100% - 240px);
    margin: 0 40px;
}

.detail table td {
    padding: 10px 0;
}

.detail table td:first-child {
    min-width: 120px;
    line-height: 20px;
    font-size: 16px;
    font-weight: 500;
    color: #222b45;
}

.detail table td:last-child {
    line-height: 20px;
    font-size: 16px;
    color: #697684;
}

.detail table ul {
    display: flex;
    flex-direction: column;
}

.detail table li {
    display: flex;
    margin-bottom: 10px;
}

.detail table li:last-child {
    margin-bottom: 0;
}

.detail table li div:first-child {
    flex: 0 0 200px;
    font-weight: 500;
}

.detail table li div {
    line-height: 25px;
}

.detail table .actors li div:first-child {
    flex: 0 0 100px;
    font-weight: 500;
}

.heading {
    top: 0;
    left: 0;
    width: 100%;
    height: 60px;
    z-index: 999;
    display: flex;
    position: fixed;
    margin-bottom: 0px;
    align-items: center;
    background-color: white;
    box-shadow: 0 4px 8px -2px rgba(0, 0, 0, 0.1);
}

.heading a {
    display: flex;
    color: #222b45;
    align-items: center;
    text-decoration: none;
}
