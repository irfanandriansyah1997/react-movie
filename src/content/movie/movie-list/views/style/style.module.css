.list {
    display: flex;
    flex-wrap: wrap;
    margin-bottom: 20px;
    justify-content: flex-start;
}

.result {
    width: 100%;
    margin-top: 30px;
}

.result b {
    font-weight: 600;
}
