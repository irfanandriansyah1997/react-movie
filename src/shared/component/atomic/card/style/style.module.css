.card {
    width: 100%;
    margin: 20px 0;
    cursor: pointer;
    border-radius: 5px;
    border: 1px solid rgba(0, 0, 0, 0.15);
    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
    transition: 0.31s cubic-bezier(0.7, 0.3, 0, 1);
    box-shadow: 0 4px 8px -2px rgba(0, 0, 0, 0.05);
}

.card:hover {
    box-shadow: 0 8px 16px -2px rgba(0, 0, 0, 0.1);
}
