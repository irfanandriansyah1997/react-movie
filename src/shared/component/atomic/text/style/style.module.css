.text {
    margin: 0;
    padding: 0;
}

.heading {
    font-size: 20px;
    line-height: 25px;
}

.default {
    font-size: 14px;
    line-height: 18px;
}

.caption {
    font-size: 12px;
    line-height: 14px;
}
