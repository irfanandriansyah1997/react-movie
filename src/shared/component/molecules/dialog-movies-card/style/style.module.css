.dialog {
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    display: block;
    position: fixed;
    opacity: 0;
    transform: translateY(100%);
    background: rgba(0, 0, 0, 0.25);
    transition: 0.31s cubic-bezier(0.7, 0.3, 0, 1);
    z-index: 9999;
}

.show {
    opacity: 1;
    transform: translateY(0);
}

.dialog div {
    top: 50%;
    left: 50%;
    width: 300px;
    overflow: hidden;
    border-radius: 5px;
    position: absolute;
    background-color: white;
    transform: translate(-50%, -50%);
}

.dialog span {
    position: absolute;
    width: 40px;
    height: 40px;
    background: #fff;
    border-radius: 50%;
    display: flex;
    align-items: center;
    justify-content: center;
    right: 5px;
    top: 5px;
    cursor: pointer;
}
