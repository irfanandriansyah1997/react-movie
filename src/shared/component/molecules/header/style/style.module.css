.firstRow {
    background-color: #333f48;
    width: 100%;
    height: 10px;
}

.secondRow {
    top: 0;
    width: 100%;
    height: 80px;
    z-index: 999;
    position: sticky;
    margin-bottom: 0px;
    background-color: white;
    box-shadow: 0 4px 8px -2px rgba(0, 0, 0, 0.1);
}

.input {
    height: 80px;
    padding: 15px 10px;
    position: relative;
}

.input label {
    top: 15px;
    bottom: 15px;
    display: flex;
    color: #697684;
    padding: 0 10px;
    position: absolute;
    align-items: center;
}

.input label span {
    font-size: 24px;
}

.input input {
    width: 100%;
    height: 50px;
    padding: 10px;
    font-size: 16px;
    color: #697684;
    padding-left: 40px;
    border-radius: 5px;
    letter-spacing: 0.5px;
    font-family: 'Avenir Next', sans-serif;
    border: 1px solid rgba(0, 0, 0, 0.15);
    transition: 0.31s cubic-bezier(0.7, 0.3, 0, 1);
}

.input input:focus {
    border: 1px solid rgba(0, 0, 0, 0.25);
}

.input button {
    top: 20px;
    border: 0;
    right: 15px;
    bottom: 20px;
    color: #fff;
    padding: 0 20px;
    position: absolute;
    background: #333f48;
    font-family: 'Avenir Next', sans-serif;
    border-radius: 5px;
}

.input button:hover {
    background: #20282e;
}
