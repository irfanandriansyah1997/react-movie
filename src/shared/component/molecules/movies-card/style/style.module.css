.card {
    width: calc(calc(100% / 4) - 20px);
    margin: 60px 10px 0;
}

.card:hover .image {
    box-shadow: 0 16px 32px -2px rgba(0, 0, 0, 0.25);
}

.container {
    display: flex;
    min-height: 200px;
    position: relative;
    align-items: center;
    flex-direction: column;
    justify-content: flex-start;
}

.content {
    width: 100%;
    display: flex;
    padding: 10px;
    text-align: center;
    flex-direction: column;
    text-decoration: none;
}

.image {
    height: 295px;
    object-fit: cover;
    margin-top: -30px;
    border-radius: 5px;
    width: calc(100% - 30px);
    transition: 0.31s cubic-bezier(0.7, 0.3, 0, 1);
}

.list {
    display: flex;
    align-items: center;
    margin: 10px -5px 5px;
}

.title {
    display: flex;
    line-height: 25px;
    align-items: center;
    justify-content: center;
    width: 100%;
    height: 64px;
    overflow: hidden;
}

.title p {
    overflow: hidden;
    max-height: 50px;
}

.listItem {
    margin: 0 5px;
    padding: 5px 10px;
    border-radius: 20px;
    border: 1px dashed rgba(0, 0, 0, 0.15);
}
