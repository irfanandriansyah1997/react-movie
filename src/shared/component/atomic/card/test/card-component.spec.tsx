import React from 'react';
import { mount, render, shallow } from 'enzyme';

import CardComponent from '../card.component';
import style from '../style/style.module.css';

describe('Testing <CardComponent> in atomic component ', () => {
    it('Should render card component correctly', () => {
        const card = render(
            <CardComponent className="ui-atomic-card">Card content</CardComponent>
        );

        const cardWrapper = shallow(
            <CardComponent className="ui-atomic-card">Card content</CardComponent>
        );

        expect(card.hasClass('ui-atomic-card')).toBe(true);
        expect(card.hasClass(style.card)).toBe(true);
        expect(cardWrapper.children()).toHaveLength(1);
    });

    it('Card Enable Click', () => {
        const spy = jest.fn();
        const card = mount(
            <CardComponent onClick={spy}>Card content</CardComponent>
        );

        card.simulate('click');
        expect(spy).toHaveBeenCalledTimes(1);
    });

    it('Generate card snapshot', () => {
        const card = mount(
            <CardComponent className="testing-generate-class-name">
                <p>Hai</p>
                <span>Selamat Data</span>
            </CardComponent>
        );

        expect(card.html()).toMatchSnapshot();
    });
});
