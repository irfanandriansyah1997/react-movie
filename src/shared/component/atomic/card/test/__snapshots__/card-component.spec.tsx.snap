// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <CardComponent> in atomic component  Generate card snapshot 1`] = `"<div class=\\"testing-generate-class-name card\\"><p>Hai</p><span>Selamat Data</span></div>"`;
